week-1-ts-ecosystem/w01e04-light-me-up/src/powerSource.ts

```diff
@@ -0,0 +1,16 @@
+export class PowerSource {
+  private energySupply = 100;
+
+  consume(energy: number) {
+    this.energySupply -= energy;
+    if (this.energySupply < 0) {
+      this.energySupply = 0;
+      throw new Error('Not enough energy');
+    }
+  }
+
+  resetGenerator() {
+    this.energySupply = 100;
```

Comment:
Jeśli `100` jako pozycję startową traktujesz ponownie. To może warto `100` przenieść do jakiejś stałej wyżej, żeby zlikwidować magic-number.

Np.
```
const BASE_ENERGY = 100;
```